pkg/partmgr/partmgr.go

```diff
@@ -18,123 +18,201 @@ under the Apache 2.0 license is conditioned upon your compliance with
+	newMngr := &PartitionManager{cfg: cfg, path: path, cyclic: false, container: cont, currentPartitionInterval: currentPartitionInterval}
+	for _, part := range cfg.Partitions {
+		path := fmt.Sprintf("%s/%d/", newMngr.path, part.StartTime/1000)
+		newPart := NewDBPartition(newMngr, part.StartTime, path)
+		newMngr.partitions = append(newMngr.partitions, newPart)
+		if newMngr.headPartition == nil {
+			newMngr.headPartition = newPart
+		} else {
+			if newMngr.headPartition.startTime < newPart.startTime {
+				newMngr.headPartition = newPart
+			}
+		}
+	}
 	return newMngr
 }
 
 // Create and Init a new Partition
-func NewDBPartition(pmgr *PartitionManager) *DBPartition {
+func NewDBPartition(pmgr *PartitionManager, startTime int64, path string) *DBPartition {
+	rollupTime, _ := utils.Str2duration(pmgr.cfg.PartitionSchemaInfo.AggregatorsGranularity)
+	partitionInterval, _ := utils.Str2duration(pmgr.cfg.PartitionSchemaInfo.PartitionerInterval)
+	chunkInterval, _ := utils.Str2duration(pmgr.cfg.PartitionSchemaInfo.ChunckerInterval)
 	newPart := DBPartition{
-		manager:       pmgr,
-		path:          pmgr.path + "/0/", // TODO: format a string based on id & format
-		partID:        1,
-		startTime:     0,
-		days:          pmgr.cfg.DaysPerObj,
-		hoursInChunk:  pmgr.cfg.HrInChunk,
-		prefix:        "",
-		retentionDays: pmgr.cfg.DaysRetention,
-		rollupTime:    int64(pmgr.cfg.RollupMin) * 60 * 1000,
+		manager:           pmgr,
+		path:              path,
+		startTime:         startTime,
+		partitionInterval: partitionInterval,
+		chunkInterval:     chunkInterval,
+		prefix:            "",
+		retentionDays:     pmgr.cfg.PartitionSchemaInfo.SampleRetention,
+		rollupTime:        rollupTime,
 	}
 
-	aggrType, _ := aggregate.AggrsFromString(pmgr.cfg.DefaultRollups) // TODO: error check & load part data from schema object
+	aggrType, _ := aggregate.AggrsFromString(pmgr.cfg.PartitionSchemaInfo.Aggregators)
 	newPart.defaultRollups = aggrType
-	if pmgr.cfg.RollupMin != 0 {
-		newPart.rollupBuckets = pmgr.cfg.DaysPerObj * 24 * 60 / pmgr.cfg.RollupMin
+	if rollupTime != 0 {
+		newPart.rollupBuckets = int(math.Ceil(float64(partitionInterval) / float64(rollupTime)))
 	}
 
 	return &newPart
 }
 
 type PartitionManager struct {
-	mtx           sync.RWMutex
-	path          string
-	cfg           *config.DBPartConfig
-	headPartition *DBPartition
-	cyclic        bool
-	ignoreWrap    bool
+	mtx                      sync.RWMutex
+	path                     string
+	cfg                      *config.Schema
+	headPartition            *DBPartition
+	partitions               []*DBPartition
+	cyclic                   bool
+	container                *v3io.Container
+	currentPartitionInterval int64 //TODO update on schema changes
 }
 
 func (p *PartitionManager) IsCyclic() bool {
 	return p.cyclic
 }
 
-func (p *PartitionManager) GetConfig() *config.DBPartConfig {
+func (p *PartitionManager) GetPartitions() []*DBPartition {
```

Comment:
You can make it exportable if you want allow others to read and modify the value.